File: backend/ProductManagementSystem.API/Services/Implementations/CategoryService.cs
using ProductManagementSystem.API.Common;
using ProductManagementSystem.API.Models;
using ProductManagementSystem.API.Repositories.Interfaces;
using ProductManagementSystem.API.Services.Interfaces;

namespace ProductManagementSystem.API.Services.Implementations
{
    public class CategoryService: ICategoryService
    {
        private readonly ICategoryRepository categoryRepository;
        public CategoryService(ICategoryRepository _categoryRepository)
        {
            categoryRepository = _categoryRepository;
        }

        public async Task<Result<List<Category>>> GetCategories()
        {
            var categories = await categoryRepository.GetCategories();
            var responseToSend = Result<List<Category>>.Ok(categories, 200);
            return responseToSend;
        }
    }
}

File: frontend/src/app/services/category/category.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Category } from '../../models/category/category.model';
import { Result } from '../../models/result/result.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class CategoryService {
  private http = inject(HttpClient);
  private URL = 'https://localhost:7088/api';

  getCategories(): Observable<Result<Category[]>> {
    return this.http.get<Result<Category[]>>(`${this.URL}/categories`);
  }
}
